R/sapflow-Rs.Rmd

```diff
@@ -72,56 +72,99 @@ read_csv("../SERC/baseliner_data/control_Kest_yr.csv",
 
 # Load weather data
 read_csv("../SERC/met_data/SECPRE_30min.csv") %>% 
-  mutate(Timestamp = ymd_hms(endDateTime)) %>% 
+  mutate(Timestamp = ymd_hms(endDateTime, tz = "UTC")) %>% 
   rename(Precip = secPrecipBulk) %>% 
   select(Timestamp, Precip) -> precip
 
 read_csv("../SERC/met_data/RH_30min.csv") %>% 
   filter(horizontalPosition == "000", verticalPosition == "060") %>% 
-  mutate(Timestamp = ymd_hms(endDateTime)) %>% 
+  mutate(Timestamp = ymd_hms(endDateTime, tz = "UTC")) %>% 
   rename(RH = RHMean, Tair = tempRHMean) %>% 
   select(Timestamp, RH, Tair) -> temprh
 
 read_csv("../SERC/met_data/PARPAR_30min.csv") %>% 
-  mutate(Timestamp = ymd_hms(endDateTime), DOY = yday(Timestamp), Date = date(Timestamp)) %>% 
-  rename(PAR = PARMean) %>% 
   filter(verticalPosition == "050") %>% 
-  select(Timestamp, DOY, PAR, Date) %>%
+  mutate(Timestamp = ymd_hms(endDateTime, tz = "UTC")) %>% 
+  select(Timestamp, PAR = PARMean) %>%
   left_join(precip, by = "Timestamp") %>% 
   left_join(temprh, by = "Timestamp") %>% 
   mutate(Timestamp = round_date(Timestamp, unit = "hour")) %>% 
   group_by(Timestamp) %>% 
-  summarise(DOY = mean(DOY), PAR = mean(PAR), 
-            Precip = mean(Precip), RH = mean(RH), Tair = mean(Tair), Date = mean(Date)) -> wx_dat
+  summarise(PAR = mean(PAR), 
+            Precip = mean(Precip), 
+            RH = mean(RH),
+            Tair = mean(Tair)) %>% 
+  mutate(DOY = yday(Timestamp), 
+         Date = date(Timestamp)) -> wx_dat_no_dpar
 
 # Compute daytime PAR as a new column
-# To split PAR groups, we need to use daily averages of daytime values
-sunlight_times <- getSunlightTimes(wx_dat$Date, lat = 38.9, lon = -77, tz = "UTC", keep = c("sunrise", "sunset"))
-
-wx_dat$sunrise <- sunlight_times$sunrise
-wx_dat$sunset <- sunlight_times$sunset
-# Check time zones before this step
-wx_dat %>% 
-  filter(Timestamp >= sunrise, Timestamp <= sunset) %>% 
+sunlight_times <- getSunlightTimes(wx_dat_no_dpar$Date,
+                                   lat = 38.9, lon = -77, tz = "EST", 
+                                   keep = c("sunrise", "sunset"))
+wx_dat_no_dpar$sunrise <- sunlight_times$sunrise
+wx_dat_no_dpar$sunset <- sunlight_times$sunset
+
+wx_dat_no_dpar %>% 
+  filter(Timestamp >= sunrise, Timestamp <= sunset) %>%   # daytime only
   group_by(DOY) %>% 
   summarise(Daytime_PAR = mean(PAR, na.rm = TRUE)) %>% 
-  right_join(wx_dat, by = "DOY")-> x
+  # we've computed mean *daytime* PAR; now merge back in
+  right_join(wx_dat_no_dpar, by = "DOY") %>% 
+  # compute daytime PAR groups - cloudy, medium, sunny
+  group_by(month(Timestamp)) %>% 
+  mutate(PAR_group = as.factor(ntile(Daytime_PAR, 3))) %>% 
+  ungroup() %>% 
+  select(-`month(Timestamp)`) ->  # no longer needed
+  wx_dat
+```
 
+## Diagnostic plots for weather data
+
+```{r, weather-qaqc}
+# Monthly averages by hour for each variable
+wx_dat %>% 
+  gather(var, value, PAR, Precip, RH, Tair) %>% 
+  mutate(Timestamp = with_tz(Timestamp, "EST"), 
```

Comment:
probably remove this line since we will do this above

Reply:
Done